pkg/k8sclient/client.go

```diff
@@ -95,31 +96,28 @@ func (c *resourceClientFactory) GetResourceClient(apiVersion, kind, namespace st
 		Kind:    kind,
 	}
 
-	client, err := c.clientPool.ClientForGroupVersionKind(gvk)
-	if err != nil {
-		return nil, "", fmt.Errorf("failed to get client for GroupVersionKind(%s): %v", gvk.String(), err)
-	}
-	resource, err := apiResource(gvk, c.restMapper)
+	gvr, err := gvkToGVR(gvk, c.restMapper)
 	if err != nil {
 		return nil, "", fmt.Errorf("failed to get resource type: %v", err)
 	}
-	pluralName := resource.Name
-	resourceClient := client.Resource(resource, namespace)
+	pluralName := gvr.Resource
+
+	resourceClient := c.dynamicClient.Resource(*gvr).Namespace(namespace)
 	return resourceClient, pluralName, nil
 }
 
-// apiResource consults the REST mapper to translate an <apiVersion, kind, namespace> tuple to a metav1.APIResource struct.
-func apiResource(gvk schema.GroupVersionKind, restMapper *discovery.DeferredDiscoveryRESTMapper) (*metav1.APIResource, error) {
+// apiResource consults the REST mapper to translate an <apiVersion, kind, namespace> tuple to a GroupVersionResource
+func gvkToGVR(gvk schema.GroupVersionKind, restMapper *restmapper.DeferredDiscoveryRESTMapper) (*schema.GroupVersionResource, error) {
 	mapping, err := restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get the resource REST mapping for GroupVersionKind(%s): %v", gvk.String(), err)
 	}
-	resource := &metav1.APIResource{
-		Name:       mapping.Resource,
-		Namespaced: mapping.Scope == meta.RESTScopeNamespace,
-		Kind:       gvk.Kind,
+	gvr := &schema.GroupVersionResource{
+		Group:    gvk.Group,
+		Version:  gvk.Version,
+		Resource: mapping.Resource.String(),
 	}
-	return resource, nil
+	return gvr, nil
```

Comment:
After testing it manually I found the issue.
`mapping.Resource` is of type `schema.GroupVersionResource` and not of type string.

Replace lines 115-120:
```Go
gvr := &schema.GroupVersionResource{
		Group:    gvk.Group,
		Version:  gvk.Version,
		Resource: mapping.Resource.String(),
	}
	return gvr, nil
```
with:
```Go
return &mapping.Resource, nil
```

Reply:
thanks, fixed, works for me locally as well now